### ISSUE TITLE
maxwell very slow, rows arrive to Maxwell from the binlog connector is low 

### ISSUE THREAD

**Issue by zengyongping** [zendesk/maxwell#1296]
I counter the problem，maxwell is very slow replicate is lag ，the rate at which rows arrive to Maxwell from the binlog connector max = 600row/s，how can I improve the  @osheroff 
![WechatIMG46](https://user-images.githubusercontent.com/8209729/61706308-4f538600-ad7a-11e9-9321-ede75acaeb9a.png)

**Comment by zengyongping** [comment]
> I counter the problem，maxwell is very slow replicate is lag ，the rate at which rows arrive to Maxwell from the binlog connector max = 600row/s，how can I improve the @osheroff
> ![WechatIMG46](https://user-images.githubusercontent.com/8209729/61706308-4f538600-ad7a-11e9-9321-ede75acaeb9a.png)

I found

**Comment by zengyongping** [comment]
> I counter the problem，maxwell is very slow replicate is lag ，the rate at which rows arrive to Maxwell from the binlog connector max = 600row/s，how can I improve the @osheroff
> ![WechatIMG46](https://user-images.githubusercontent.com/8209729/61706308-4f538600-ad7a-11e9-9321-ede75acaeb9a.png)

**Comment by dbapower** [comment]
+1,is the Split server roles option cause the reason?@osheroff

**Comment by dbapower** [comment]
+1,
the mysql slave's lag is 100-200s at 1:00 am,after 1:10 am there is no lag in slave .
Til now at 7:00 pm, maxwell has just catch up the binlog position at 4:00 am's .

**Comment by dbapower** [comment]
+1,
the mysql slave's lag is 100-200s at 01:00, after 01:10  there is no lag in slave .
Til now at 17:00 , maxwell has just catch up the binlog position at 04:00 .

**Comment by dbapower** [comment]
we already try the latest version of maxwell,still have the problem.
in order to reduce the impact of production DB,we use the Split server roles,but the store captured schema DB is ok ,both network,mem and disk IO is all right.
And we use the maxwell's filter function.
So,is the  Split server roles or filter cause the lag ?
@osheroff
thanks.

**Comment by osheroff** [comment]
JavaScript filter or regular filter?  What producer are you using?  Most of the time the producer is the bottleneck.

**Comment by liquidgenius** [comment]
@dbapower I am interested in how you've set up your config.properties and what permissions on the MySQL you've set for your split server roles. Would you mind sharing them?

**Comment by dbapower** [comment]
@osheroff we use the maxwell's filter, kafka,@liquidgenius
here is the config.properties:

kafka.request.timeout.ms=300000
# tl;dr config
ignore_producer_error=false
log_level=debug
kafka_version=0.9.0.1
producer=kafka
kafka.bootstrap.servers=xx.xx.xx.140:39092


client_id=maxwell_xx.xx.xx.224_3993
kafka_topic=maxwell_xx.xx.xx.224_3993
replication_host=xx.xx.xx.41
replication_port=3993
replica_server_id=8091
filter= exclude: *.*, include: lcs.ln_loan,include: lcs.ln_plan,include: lcs.ln_loan_stats,include: lcs.pd_product_def, include: lcs.tr_tran_proc_db,include: lcs.tr_tran_proc_rp
http_port=8091

# mysql login info
host=xx.xx.xx.226
user=maxwell
password=xxxxxxx
port=3306
schema_database=maxwell_prod

replication_user=maxwell
replication_password=xxxxx

output_binlog_position=true
output_ddl=true

kafka.compression.type=snappy
kafka.retries=0
kafka.acks=1

producer_partition_by=table

metrics_type=slf4j,http
# The prefix maxwell will apply to all metrics
metrics_prefix=MaxwellMetrics # default MaxwellMetrics

# Enable (dropwizard) JVM metrics, default false
metrics_jvm=false

# When metrics_type includes slf4j this is the frequency metrics are emitted to the log, in seconds
metrics_slf4j_interval=60
# When metrics_type includes http or diagnostic is enabled, this is the port the server will bind to.

thanks.

**Comment by dbapower** [comment]
@osheroff we use the maxwell's basic filter, producer is kafka 0.9.0.1,@liquidgenius
here is the config.properties:

kafka.request.timeout.ms=300000

ignore_producer_error=false
log_level=debug
kafka_version=0.9.0.1
producer=kafka
kafka.bootstrap.servers=xx.xx.xx.140:39092

client_id=maxwell_xx.xx.xx.224_3993
kafka_topic=maxwell_xx.xx.xx.224_3993
replication_host=xx.xx.xx.41
replication_port=3993
replica_server_id=8091
filter= exclude: ., include: lcs.ln_loan,include: lcs.ln_plan,include: lcs.ln_loan_stats,include: lcs.pd_product_def, include: lcs.tr_tran_proc_db,include: lcs.tr_tran_proc_rp
http_port=8091
host=xx.xx.xx.226
user=maxwell
password=xxxxxxx
port=3306
schema_database=maxwell_prod
replication_user=maxwell
replication_password=xxxxx
output_binlog_position=true
output_ddl=true
kafka.compression.type=snappy
kafka.retries=0
kafka.acks=1
producer_partition_by=table
metrics_type=slf4j,http
metrics_prefix=MaxwellMetrics # default MaxwellMetrics
metrics_jvm=false
metrics_slf4j_interval=60

thanks.





Production DB(replication_host ) of maxwell privileges is :
SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON *.*
Store the captured schema's DB(host) of maxwell privileges is:

**Comment by dbapower** [comment]
@osheroff we use the maxwell's basic filter, producer is kafka 0.9.0.1,@liquidgenius
here is the config.properties:

kafka.request.timeout.ms=300000

ignore_producer_error=false
log_level=debug
kafka_version=0.9.0.1
producer=kafka
kafka.bootstrap.servers=xx.xx.xx.140:39092

client_id=maxwell_xx.xx.xx.224_3980
kafka_topic=maxwell_xx.xx.xx.224_3980
replication_host=xx.xx.xx.41
replication_port=3980
replica_server_id=8091
filter= exclude: ., include: lcs.ln_loan,include: lcs.ln_plan,include: lcs.ln_loan_stats,include: lcs.pd_product_def, include: lcs.tr_tran_proc_db,include: lcs.tr_tran_proc_rp
http_port=8091
host=xx.xx.xx.226
user=maxwell
password=xxxxxxx
port=3306
schema_database=maxwell_prod
replication_user=maxwell
replication_password=xxxxx
output_binlog_position=true
output_ddl=true
kafka.compression.type=snappy
kafka.retries=0
kafka.acks=1
producer_partition_by=table
metrics_type=slf4j,http
metrics_prefix=MaxwellMetrics # default MaxwellMetrics
metrics_jvm=false
metrics_slf4j_interval=60

thanks.

**Comment by dbapower** [comment]
@liquidgenius 
Production DB(replication_host ) of maxwell privileges is :
SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON *.*
Store the captured schema's DB(host) of maxwell privileges is:
ALL on maxwell_prod.*

**Comment by dbapower** [comment]
Production DB(replication_host ) of maxwell privileges is :
SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON *.*
Store the captured schema's DB(host) of maxwell privileges is:
ALL ON maxwell_prod.*

**Comment by dbapower** [comment]
@liquidgenius
Production DB(replication_host ) of maxwell privileges is :
SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON `*.*`
Store the captured schema's DB(host) of maxwell privileges is:
ALL ON maxwell_prod.*

**Comment by liquidgenius** [comment]
@dbapower I don't have much expertise, but do you have any improvement if you set the debug level to INFO, and turn metrics off?

Also I notice that the parameter http_port is listed twice in the config.

Thank you for the example, I found it extremely helpful in understanding how split role servers are set up.

**Comment by liquidgenius** [comment]
@dbapower It also appears as if you may be running CDC from all tables through the same Kafka topic? kafka_topic=maxwell_xx.xx.xx.224_3980

What about setting up individual topics for each table and using the following template to send CDC for each table trgough its own topic?
kafka_topic=testing_%{database}_%{table}

**Comment by dbapower** [comment]
@@osheroff  thank you so much!bottleneck is kafka,we changed another kafka cluster, the speed is ok!
